sdk/data/azcosmos/example_test.go

```diff
@@ -501,7 +501,7 @@ func ExampleContainerClient_ReplaceItem_optimisticConcurrency() {
 		// Replace with Etag
 		etag := itemResponse.ETag
 		itemResponse, err = container.ReplaceItem(context.Background(), pk, id, marshalledReplace, &azcosmos.ItemOptions{IfMatchEtag: &etag})
-		var httpErr azcore.HTTPResponse
+		var httpErr interface{ RawResponse() *http.Response }
```

Comment:
Given that your error type already exposes the raw *http.Response via a field, I think this can just go away.  Previously, we exposed this interface on error types so that callers had a way to get the underlying response if they cared about it.

Reply:
Is our error type still correct in terms of the fields it exposes related to the recommendations? Or should we do it differently?

Reply:
Our contract for error types is that they must expose the following bits of data.
- the error code, usually from `x-ms-error-code` header or maybe some field in the response body
- the HTTP status code
- the raw response

Anything else should be contextual to help diagnose the error and come from the `Error()` method.  For our code-generated SDKs, we have [`azcore.ResponseError`](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/internal/shared/response_error.go#L90-L99) for this.

Is `cosmosError` the common error type that's returned for failed API calls?

Reply:
Yes, for failed API calls we return `cosmosError` which right now exposes (https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/azcosmos/cosmos_error.go):

* `func (e *cosmosError) StatusCode() int`
* `func (e *cosmosError) ErrorCode() string`
* `func (e *cosmosError) RawResponse() *http.Response`

Reply:
I see.  What's the public type that customers use to access this info?

Reply:
Take a look at my proposed changes to `CosmosError`.

Reply:
RIght now there was no public type, as far as I remember, our earlier discussion was that as long as our internal type worked with the proposed azcore interface, it would be good enough. There is no more azcore error interface now?

Reply:
I don't mind making CosmosError public as long as it's fine from the API design perspective and follows the azcore guidelines 😄

Reply:
@JeffreyRichter this should align with our current thinking for error handling.  Any concerns?

Reply:
I think CosmosError should be removed entirely and the azcore.ResponseError shoud be used instead.

Reply:
We could do that.  However, I want to point out that having `CosmosError` would allow for any Cosmos-specific error info to be exposed in the future.  How likely is this to happen, and more generally, do we want to support this scenario?

Reply:
Another option would be to have `CosmosError` wrap an `azcore.ResponseError` instead of reimplementing it.  That way we have consistent handling while keeping the extensibility for Cosmos-specific content.

Reply:
I'll also add, that if we go with the wrapping technique, it means that we could add `CosmosError` later if/when needed.

Reply:
Fixed.